packages/web3-eth-abi/src/api/functions_api.ts

```diff
@@ -0,0 +1,41 @@
+import { AbiError } from 'web3-common';
+import { sha3Raw } from 'web3-utils';
+import { isAbiFunctionFragment, jsonInterfaceMethodToString } from '../utils';
+import { JsonAbiFunctionFragment } from '../types';
+import { encodeParameters } from './parameters_api';
+
+/**
+ * Encodes the function name to its ABI representation, which are the first 4 bytes of the sha3 of the function name including  types.
+ */
+export const encodeFunctionSignature = (functionName: string | JsonAbiFunctionFragment): string => {
+	if (typeof functionName !== 'string' && !isAbiFunctionFragment(functionName)) {
+		throw new Error('Invalid parameter value in encodeFunctionSignature');
+	}
+
+	let name: string;
+
+	if (typeof functionName === 'function' || (typeof functionName === 'object' && functionName)) {
+		name = jsonInterfaceMethodToString(functionName);
+	} else {
+		name = functionName;
+	}
+
+	return sha3Raw(name).slice(0, 10);
+};
+
+/**
+ * Encodes a function call from its json interface and parameters.
+ */
+export const encodeFunctionCall = (
+	jsonInterface: JsonAbiFunctionFragment,
+	params?: unknown[],
```

Comment:
Its not optional. In case of no param user should pass empty array. https://web3js.readthedocs.io/en/v1.5.2/web3-eth-abi.html#encodefunctioncall as per 1.x api